ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/Proxy.kt

```diff
@@ -1,27 +1,41 @@
 /*
- * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
 package io.ktor.client.tests.utils.tests
 
 import io.ktor.client.utils.*
 import io.ktor.http.*
+import io.ktor.network.selector.*
 import io.ktor.network.sockets.*
 import io.ktor.util.*
 import io.ktor.utils.io.*
+import kotlinx.coroutines.*
+import kotlin.coroutines.*
+import kotlin.time.*
 
 public suspend fun proxyHandler(socket: Socket) {
     val input = socket.openReadChannel()
     val output = socket.openWriteChannel()
 
-    val statusLine = input.readUTF8Line()
+    var statusLine = input.readUTF8Line()
     val requestData = StringBuilder()
     requestData.append(statusLine).append("\n")
     while (true) {
         val line = input.readUTF8Line() ?: ""
         requestData.append(line).append("\n")
         if (line.isEmpty()) {
```

Comment:
Please consider inverting if condition and use an early break.

Reply:
done